packages/coin-tester/src/components/coins/algo/index.tsx

```diff
@@ -0,0 +1,351 @@
+const algodPort = 4001;
+let algodClient = new algosdk.Algodv2(algodToken, algodServer, algodPort);
+
+const CoinAlgoPage: FC<Props> = (props: Props) => {
+  const coinALGO = new CoinALGO();
+  const disabled = !props.transport || props.isLocked;
+
+  const [payload, setPayload] = useState('');
+  const [result, setResult] = useState('');
+  const [address, setAddress] = useState('');
+  const [receiver, setReceiver] = useState('5B3X56E3KVGS3D5263AWYWMUFBJUX7IZ3OYBUYVH6AB4ZDNJSGT4MQAG2U');
+  const [signature, setSignature] = useState('');
+  const [amount, setAmount] = useState('100000');
+  const [assetAmount, setAssetAmount] = useState('0');
+  const [assetId, setAssetId] = useState('82200197');
+  const [appIndex, setAppIndex] = useState('82200197');
+  const [assetName, setAssetName] = useState('coolbitx');
+  const [total, setTotal] = useState("1000");
+  const [freezeAddress, setFreezeAddress] = useState('5B3X56E3KVGS3D5263AWYWMUFBJUX7IZ3OYBUYVH6AB4ZDNJSGT4MQAG2U');
+  const [freezeAssetId, setFreezeAssetId] = useState('82200197');
+
+  const getAddress = () => {
+    useRequest(async () => {
+      const appId = localStorage.getItem('appId');
+      if (!appId) throw new Error('No Appid stored, please register!');
+      return coinALGO.getAddress(props.transport!, props.appPrivateKey, appId);
+    }, props).then(setAddress);
+  };
+
+  const stringToBytes = (value: any) => {
+    return new Uint8Array(Buffer.from(value));
+  }
+
+  const signTransaction = async (transaction: any) => {
+    const appId = localStorage.getItem('appId');
+    if (!appId) throw new Error('No Appid stored, please register!');
+
+    const signTxData = {
+      transport: props.transport!,
+      appPrivateKey: props.appPrivateKey,
+      appId,
+      transaction: transaction,
+      addressIndex: 0,
+    };
+
+    return coinALGO.signTransaction(signTxData);
+  }
+
+  const getTransaction = async () => {
+    try {
+      const signed = await apdu.tx.getSignedHex(props.transport!);
+      setPayload(signed.signedTx);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const sendTransaction = async () => {
+    const signedTransaction = Uint8Array.from(Buffer.from(signature, 'hex'))
+    setResult((await algodClient.sendRawTransaction(signedTransaction).do()).txId);
+  }
+
+  const signPaymentTransaction = async () => {
+    let params = await algodClient.getTransactionParams().do();
+    const enc = new TextEncoder();
+    const note = enc.encode("Payment Transaction");
+
+    useRequest(async () => {
+      let transactionObject = algosdk.makePaymentTxnWithSuggestedParamsFromObject({
+        from: address,
+        to: receiver,
+        amount: Number(amount),
+        note: note,
+        suggestedParams: params
+      });
+
+      const transactionForSDK = transactionObject.get_obj_for_encoding();
+      console.log("Transaction Object : ", transactionForSDK);
+
+      return signTransaction(transactionForSDK);
```

Comment:
It will be good if we can have a `signPaymentTransaction` function in `@coolwallet/algo`.
Developers should not import `algosdk`  to use `@coolwallet/algo` for signing transaction proposed.